src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapSessionOptions.Windows.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+using System.Net;
+using System.Security.Cryptography.X509Certificates;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+using System.Collections;
+using System.Text;
+using System.Diagnostics;
+using System.Security.Authentication;
+using System.Runtime.CompilerServices;
+
+namespace System.DirectoryServices.Protocols
+{
+    public partial class LdapSessionOptions
+    {
+        private static void PALCertFreeCRLContext(IntPtr certPtr) => Wldap32.CertFreeCRLContext(certPtr);
+
+        internal bool FQDN
+        {
+            set
+            {
+                // set the value to true
+                SetIntValueHelper(LdapOption.LDAP_OPT_AREC_EXCLUSIVE, 1);
```

Comment:
This is... bad.

* Don't have a set-only property.
* A property setter should use its value.


Perhaps this wants to be `SetFqdnRequired()`?

Reply:
I agree it's wrong, will change to method. For reference, it was like this because we were only setting this property once in our code, and it was always setting it to true.